### PR TITLE
port deep parameter logging from Badiapp/grape_logging

### PR DESCRIPTION
**Pull request by jason-uh**
This ports deep parameter logging from Badiapp/grape_logging without having to depend upon Rails directly (lifts the `ParameterFilter` from Rails, or uses it indirectly if Rails is available). I also started to add some specs.

### CODE REVIEW

**Comment by jason-uh** [comment]
@aserafin Is there any chance of getting this PR merged and a new gem built?

File: lib/grape_logging/loggers/filter_parameters.rb [hunk 1]
```diff
@@ -1,23 +1,32 @@
...
-        parameters
+
+      def parameter_filter
+        @parameter_filter ||= ParameterFilter.new(@replacement, @filter_parameters)
+      end
+
+      def safe_parameters(request)
+        # Now this logger can work also over Rails requests
+        return clean_parameters(request.env[AD_PARAMS] || {}) if request.params.empty?
+        clean_parameters(request.params.clone)
```
Comment (aserafin) [inline lib/grape_logging/loggers/filter_parameters.rb]: looking at `ParameterFilter` class I don't think we need to clone parameters anymore, right?

Comment (aserafin) [inline lib/grape_logging/loggers/filter_parameters.rb]: could we also do `if ... else ... end` here?

**Comment by aserafin** [comment]
hey @jason-uh - sorry it took so long and thanks for contribution, especially the tests 💪 could you take a look at two notes I left?

File: lib/grape_logging/loggers/filter_parameters.rb (hunk 1 above)
Comment (jason-uh) [inline lib/grape_logging/loggers/filter_parameters.rb]: ☑️ Done and done.

**Comment by jason-uh** [comment]
Thanks @aserafin.

**Comment by jason-uh** [comment]
@aserafin would you also publish a new version of the gem? Thanks.

**Comment by aserafin** [comment]
@jason-uh it's live https://rubygems.org/gems/grape_logging/versions/1.4.0 🎉

**Comment by jason-uh** [comment]
Awesome @aserafin. Thanks a bunch, and thanks to @oricodes89 from @badiapp for the initial code.

**Comment by oricodes89** [comment]
Thanks to @dsantosmerino! @badiapp is happy to help.

**Pull request merged**